### PR TITLE
update user event usage in specs

### LINKED ISSUE #394: consider using userEvent rather than fireEvent
[according to the docs](https://testing-library.com/docs/dom-testing-library/api-events)
> Most projects have a few use cases for fireEvent, but the majority of the time you should probably use [@testing-library/user-event](https://testing-library.com/docs/user-event/intro)

Here's an updated version of `00-single-component.spec.ts`:
``` typescript
import { render, screen } from '@testing-library/angular';
import userEvent from '@testing-library/user-event';

import { SingleComponent } from './00-single-component';

test('renders the current value and can increment and decrement', async () => {
  const user = userEvent.setup()
  await render(SingleComponent);

  const incrementControl = screen.getByRole('button', { name: /increment/i });
  const decrementControl = screen.getByRole('button', { name: /decrement/i });
  const valueControl = screen.getByTestId('value');

  expect(valueControl).toHaveTextContent('0');

  await user.click(incrementControl);
  await user.click(incrementControl);
  expect(valueControl).toHaveTextContent('2');

  await user.click(decrementControl);
  expect(valueControl).toHaveTextContent('1');
});
```

### PR DESCRIPTION
**Pull request by markgoho**
this closes #394

### CODE REVIEW

File: .githooks/pre-commit
Comment (markgoho) [inline .githooks/pre-commit:1]: this file was not natively executable so I ran `chmod +x` on it and committed the change, I can take out if needed

Comment (timdeschryver) [reply .githooks/pre-commit:1]: That's fine, thanks!

**Review by timdeschryver — COMMENTED** [review]
It seems like some tests broke with this, could you take a look at it please?

**Comment by markgoho** [comment]
> It seems like some tests broke with this, could you take a look at it please when you have the time?

So sorry about this! I've fixed all that I could in reasonable amount of time. There were a few that related to fake timers and async stuff that just would not pass.

**Review by timdeschryver — APPROVED** [review]
Thanks!

**Comment by timdeschryver** [comment]
@all-contributors please add @markgoho  for maintenance, docs

**Pull request merged**